src/python/tests/core/api/test_scheduler.py

```diff
@@ -0,0 +1,283 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+import time
+import datetime
+import json
+import asyncpg
+import requests
+import pytest
+import asyncio
+from foglamp.core.scheduler import Schedule, Task
+
+
+__author__ = "Amarendra K Sinha"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+# Module attributes
+__DB_NAME = "foglamp"
+BASE_URL = 'http://localhost:8082/foglamp'
+headers = {"Content-Type": 'application/json'}
+
+
+async def add_master_data():
+    conn = await asyncpg.connect(database=__DB_NAME)
+    await conn.execute('truncate foglamp.tasks')
+    await conn.execute(''' DELETE from foglamp.schedules WHERE process_name in ('testsleep1', 'testsleep5', 'testsleep10', 'testsleep30')''')
+    await conn.execute(''' DELETE from foglamp.scheduled_processes WHERE name in ('testsleep1', 'testsleep5', 'testsleep10', 'testsleep30')''')
+    await conn.execute('''insert into foglamp.scheduled_processes(name, script)
+        values('testsleep1', '["sleep", "1"]')''')
+    await conn.execute('''insert into foglamp.scheduled_processes(name, script)
+        values('testsleep10', '["sleep", "10"]')''')
+    await conn.execute('''insert into foglamp.scheduled_processes(name, script)
+        values('testsleep30', '["sleep", "30"]')''')
+    await conn.execute('''insert into foglamp.scheduled_processes(name, script)
+        values('testsleep5', '["sleep", "5"]')''')
+    await conn.execute('commit')
+    await asyncio.sleep(4)
+
+async def delete_master_data():
+    conn = await asyncpg.connect(database=__DB_NAME)
+    await conn.execute('truncate foglamp.tasks')
+    await conn.execute(''' DELETE from foglamp.schedules WHERE process_name in ('testsleep1', 'testsleep5', 'testsleep10', 'testsleep30')''')
+    await conn.execute(''' DELETE from foglamp.scheduled_processes WHERE name in ('testsleep1', 'testsleep5', 'testsleep10', 'testsleep30')''')
+    await conn.execute('commit')
+    await asyncio.sleep(4)
+
+
+def delete_all_schedules():
+    """
+    Deletes all schedules.
```

Comment:
please add a docstring, why do we need this? while we do have delete_master_data():